models/template_helpers.go

```diff
@@ -0,0 +1,222 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"time"
+
+	"github.com/percona-platform/saas/pkg/alert"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+)
+
+func checkUniqueTemplateName(q *reform.Querier, name string) error {
+	if name == "" {
+		panic("empty template name")
+	}
+
+	template := &Template{Name: name}
+	switch err := q.Reload(template); err {
+	case nil:
+		return status.Errorf(codes.AlreadyExists, "Template with name %q already exists.", name)
+	case reform.ErrNoRows:
+		return nil
+	default:
+		return errors.WithStack(err)
+	}
+}
+
+// FindTemplates returns saved notification rule templates.
+func FindTemplates(q *reform.Querier) ([]Template, error) {
+	structs, err := q.SelectAllFrom(TemplateTable, "")
```

Comment:
nit: `structs` -> `rows`

Reply:
`structs` is consistent with other models.

https://github.com/percona/pmm-managed/blob/c2cd5d57da2f4dccfe7b2694e1f75e0f53125ebd/models/agent_helpers.go#L155

Reply:
fine